process_request.py

```diff
@@ -1,33 +1,24 @@
-import json
-
 from data.leaderboard import Leaderboard
 from data.message_item import BaseRequest
 from data.responder import Responder
 from data.test_message_item import REVOKE_FRIEND_REQUEST, GET_ACCOUNT_INFO, SAVE_ACCOUNT_INFO_BY_KEY, CHANGE_PASSWORD, \
     GET_MOST_CHESS_GAMES_WON, GET_LONGEST_WIN_STREAK, SIGNOUT, REMOVE_FRIEND, ACCEPT_FRIEND_REQUEST, \
     SEND_FRIEND_REQUEST, GET_FRIEND_REQUESTS, GET_FRIENDS_LIST, GET_USER_STATS, CREATE_ACCOUNT, SIGNIN
 from database.db import DB
-from global_logger import logger, logged_method
+from global_logger import logger, log_error
 from user.account_management import AccountManagement
 from user.friends_management import FriendsManagement
 from user.signin import Signin
 from user.validate_request import RequestValidator
 
 
-class ProcessRequest:
+class RequestProcessor:
 
     # PrecessRequest is set up to be a separate process in the OS and
     # will hold the shared request queue object. It will pull requests
     # from the queue as they are inserted from the listener
-    def __init__(self, request_queue):
-        with open('./params.json', 'r') as f:
-            data = json.loads(f.read())
-        reader = data['db_host']
-        writer = data['db_host']
-        username = data['db_username']
-        password = data['db_password']
-        db_name = data['db_name']
-        self.database = DB(username, password, reader, writer, db_name)
+    def __init__(self, request_queue, database: DB):
+        self.database = database
```

Comment:
I changed things so the DB implementation is determined before the controller object is created and injected via constructors. I did this so we are not locked into a MySQL implementation. We could even create an in-memory mock db to be used when doing some testing if we wanted.